### PR TITLE
chore: php 8 support

### PR DESCRIPTION
**Pull request by fdalcin**
Issue #2636
Part of pressbooks/private#961

This PR updates the code to support php 8

### CODE REVIEW

File: .github/workflows/standards-and-tests.yml [hunk 1]
```diff
@@ -15,25 +15,17 @@ jobs:
...
-            os: ubuntu-20.04
-            wordpress: latest
-            epubcheck: 4.2.6
-            prince: 14.3-1
-            experimental: true
+          - experimental: true
+          - experimental: false
+            php: 8.0
+            wordpress: 6.0.2
+
```
Comment (fdalcin) [inline .github/workflows/standards-and-tests.yml:28]: I've updated the pipeline a bit so now it runs the following matrix

|. PHP | WP | experimental |
| :-: | :-: | :-: |
| 7.4 | 6.0.2 | true |
| 7.4 | latest | true |
| 8.0 | 6.0.2 | false |
| 8.0 | latest | true |

This results on 4 different jobs that will run tests, however only the PHP 8.0 + WP 6.0.2 will run the CS and Code coverage. In the future when we drop PHP 7.4 support we simply remove 7.4 from the test matrix and the pipeline should be ok.

**Review by richard015ar — APPROVED** [review]
I tested it locally and works without notice, errors or warnings in the debug and error logs.
Changes LGTM.
However, more manual testing is recommended.

**Review by arzola — CHANGES_REQUESTED** [review]
LGTM

Just we need to keep an eye on the `libxml_disable_entity_loader` removal because according to the docs on versions lower than PHP 8 using the following flag could lead to a security thread

image

https://php.watch/versions/8.0/libxml_disable_entity_loader-deprecation

So I suggest to add one of the following conditions with a TODO tag to remove when we get rid of PHP 7.4

```
if (\PHP_VERSION_ID < 80000) {
      libxml_disable_entity_loader(true);
 }
//or
if (\LIBXML_VERSION < 20900) {
      libxml_disable_entity_loader(true);
 }
```

**Comment by fdalcin** [comment]
Thanks @arzola, I've addressed your suggestion

**Pull request merged**